### PR TITLE
Reuse MaybeConvertWasmExtensionConfig

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@bianpengyuan

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -616,10 +616,7 @@ func (configgen *ConfigGeneratorImpl) buildInboundFilterchains(in *plugin.InputP
 			}
 		}
 	}
-	// only inject WasmPlugins for Sidecar inbound listeners and routers outbound listeners
-	if in.Node.Type == model.SidecarProxy {
-		extension.AddWasmPluginsToMutableObjects(mutable, in.Push.WasmPlugins(in.Node))
-	}
+	extension.AddWasmPluginsToMutableObjects(mutable, in.Push.WasmPlugins(in.Node))
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:619]: PR title sounds like a refactor but this seems like a massive API change?

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:619]: Or is this function only called for sidecar already?

Comment (bianpengyuan) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:619]: Yeah I think that is the case.

**Review by bianpengyuan — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:619]: Tep, this is called by buildInboundFilterchains

**Review by howardjohn — APPROVED** [review]

**Pull request merged**